### PR TITLE
[REVIEW] Add GPU SHAP on Azure

### PR DESCRIPTION
**Pull request by Nanthini10**
- Closes https://github.com/rapidsai/cloud-ml-examples/issues/114 
- Adds a simple classification example with Higgs dataset for explainability on Azure
- Using `N_ROWS` as the dataset is huge and causes memory error for larger values
- Right now, we need to specify `interpret-community==0.18` because `azureml-interpret` uses 0.17; but going forward we can get rid of it as it'll get updated in the next release

### CODE REVIEW

**Review by akaanirban — COMMENTED** [review]
Looks good in general. Ran into some errors in the Explanation Dashboard. Did it run for `interpret_community.widget.ExplanationDashboard`?

File: azure/notebooks/remote-explanation/azure-gpu-shap.ipynb [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+    "1. Explain models with engineered features:\n",
+    "    1. [Simple feature transformations](https://github.com/interpretml/interpret-community/blob/master/notebooks/simple-feature-transformations-explain-local.ipynb)\n",
+    "    1. [Advanced feature transformations](https://github.com/interpretml/interpret-community/blob/master/notebooks/advanced-feature-transformations-explain-local.ipynb)\n",
+    "1. [Save model explanations via Azure Machine Learning Run History](../run-history/save-retrieve-explanations-run-history.ipynb)\n",
+    "1. Inferencing time: deploy a classification model and explainer:\n",
+    "    1. [Deploy a locally-trained model and explainer](../scoring-time/train-explain-model-locally-and-deploy.ipynb)\n",
+    "    1. [Deploy a locally-trained keras model and explainer](../scoring-time/train-explain-model-keras-locally-and-deploy.ipynb)\n",
+    "    1. [Deploy a remotely-trained model and explainer](../scoring-time/train-explain-model-on-amlcompute-and-deploy.ipynb)"
```
[the remark this answers is not in the document]
Comment (Nanthini10) [reply azure/notebooks/remote-explanation/azure-gpu-shap.ipynb]: Removed these!

File: azure/notebooks/remote-explanation/azure-gpu-shap.ipynb [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+    "Load the visualization dashboard"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from interpret_community.widget import ExplanationDashboard"
```
[the remark this answers is not in the document]
Comment (Nanthini10) [reply azure/notebooks/remote-explanation/azure-gpu-shap.ipynb:397]: It was the error from cudf and then the samples used in the script was 50, the new version should have that fixed.

File: azure/notebooks/remote-explanation/azure-gpu-shap.ipynb [hunk 3]
```diff
@@ -0,0 +1,463 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Check core SDK version number\n",
+    "import azureml.core\n",
```
[the remark this answers is not in the document]
Comment (Nanthini10) [reply azure/notebooks/remote-explanation/azure-gpu-shap.ipynb:26]: Good point, let me add that!

File: azure/notebooks/remote-explanation/azure-gpu-shap.ipynb (hunk 2 above)
[the remark this answers is not in the document]
Comment (akaanirban) [reply azure/notebooks/remote-explanation/azure-gpu-shap.ipynb:397]: Sound good :)

[the remark this answers is not in the document]
Comment (Nanthini10) [reply azure/notebooks/remote-explanation/azure-gpu-shap.ipynb:422]: We'll keep it as it is for now, update once interpret-community updates.

**Review by akaanirban — COMMENTED** [review]
Looks good to me. Runs fine on Rapids 0.19 with no issues. Dashboard working as well. I like the additional texts you added for explaining things.

**Comment by Nanthini10** [comment]
@zronaghi This is ready to be merged pending your review. The code works end-to-end

File: azure/notebooks/remote-explanation/azure-gpu-shap.ipynb [hunk 4]
```diff
@@ -0,0 +1,470 @@
+{
```
Comment (zronaghi) [inline azure/notebooks/remote-explanation/azure-gpu-shap.ipynb:1]: <p>I'd also link to the README for these setup steps on Azure</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cloud-ml-examples/pull/116/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (zronaghi) [inline azure/notebooks/remote-explanation/azure-gpu-shap.ipynb:1]: <p>If they download the config file from the portal, these steps are not needed. In this case, they're creating the workspace that hasn't been setup through portal. So we can either link the README to follow similar steps as other demos and use the portal or just comment it out and mention either way works</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cloud-ml-examples/pull/116/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (zronaghi) [inline azure/notebooks/remote-explanation/azure-gpu-shap.ipynb:1]: <p>Can we run any part of this demo on multi-GPU? If not, we can just recommend single GPU VMs</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/rapidsai/cloud-ml-examples/pull/116/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

File: azure/notebooks/remote-explanation/train_explain.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    joblib.dump(value=reg, filename=os.path.join(OUTPUT_DIR,
+                                                 model_file_name))
+
+# register the model
+run.upload_file('original_model.pkl', os.path.join('./outputs/', model_file_name))
+original_model = run.register_model(model_name='model_explain_model_on_amlcomp',
+                                    model_path='original_model.pkl')
+
+# Explain predictions on your local machine
+tabular_explainer = TabularExplainer(model, X_train.to_pandas(), features=X_train.columns, use_gpu=True)
```
Comment (zronaghi) [inline azure/notebooks/remote-explanation/train_explain.py:68]: Another thought for integration, can we use cuDF in the GPU explainer?

**Review by zronaghi — APPROVED** [review]
Looks great, few minor comments

File: azure/notebooks/remote-explanation/azure-gpu-shap.ipynb (hunk 4 above)
Comment (Nanthini10) [reply azure/notebooks/remote-explanation/azure-gpu-shap.ipynb:1]: You can also load an existing workspace that's what the `try` block tries to do. It skips the need to download the `config.json` file and place it in the folder. I'll link it to the README and mention these are two different ways.

Comment (Nanthini10) [reply azure/notebooks/remote-explanation/azure-gpu-shap.ipynb:1]: We cannot run it on multi-GPU, I'll use the VM 6s VM

**Review by akaanirban — COMMENTED** [review]
Approved the changes.

**Review by akaanirban — APPROVED** [review]

**Pull request merged**